packages/perseus/src/renderer-util.test.ts

```diff
@@ -743,23 +745,148 @@ describe("renderer utils", () => {
             });
         });
 
+        it("should return empty if any validator returns empty", () => {
+            // Act
+            const validatorSpy = jest
+                .spyOn(DropdownWidgetExport, "validator")
+                // Empty
+                .mockReturnValueOnce({
+                    type: "invalid",
+                    message: null,
+                })
+                // Not empty
+                .mockReturnValueOnce(null);
+            const scoringSpy = jest.spyOn(DropdownWidgetExport, "scorer");
```

Comment:
From the perspective of an outsider, I like this strategy. From the perspective of someone actively working on splitting up WidgetExports, I wonder if there's a way to do these tests without spying on WidgetExports; for example we could get 90% there by creating an empty user input object and that would keep me from having to refactor this test when I make my changes.

No action needed, just something that came to mind.

Reply:
I see your point.  However, I think that we'll have some sort of registry in the server or server-scoring package. I see the validators and scorers forever living near each other (or do validators live in `perseus-core` and scorers live in `perseus-score`.  I'm imagining you're thinking through that right now. 🤔 

I think I'll land this as is and we can revisit when the server/scoring pieces take shape further.